index.bs

```diff
@@ -3902,73 +4157,75 @@ Each {{WebRequestEvent}} has:
+          whose [=WebRequestAuthDetails/challenger=],
+          [=WebRequestAuthDetails/isProxy=],
+          [=WebRequestAuthDetails/scheme=], and
+          [=WebRequestAuthDetails/realm=] fields have the values
+          |challenger|, |isProxy|, |scheme|, and |realm| respectively.
 
       1. [=Queue a global task=] on the [=DOM manipulation task source=] of
           |embedderGlobal| that will run the following steps:
 
-          1. Let |result| be null.
+          1. [=Dispatch=] |event| to |interceptor|.
 
-          1. If |config|'s [=WebRequest handler config/asyncBlocking=] flag is
-              `true`, then:
+          1. Let |credentials| be |event|'s
+              [=WebRequestAuthRequiredEvent/authCredentials=].
 
-              1. Let |callback| be a [=function=] that takes a
-                  {{BlockingResponse}} |blockingResponse| argument and when
-                  executed will set |result| to |blockingResponse|.
+          1. If |event|'s [=WebRequestAuthRequiredEvent/cancel=] is `true`,
+              then set |result|'s [=WebRequestEvent result/cancel=] to `true`.
 
-              1. Call |config|'s [=WebRequest handler config/handler=] given
-                  |details| and |callback|.
+          1. Otherwise, if |credentials| is a {{Promise}}, then:
 
-              1. Wait [=in parallel=] for |callback| to be invoked, and then
-                  continue executing these steps.
+              1. Let |signal| be |event|'s [=WebRequestAuthRequiredEvent/
+                  options=]["{{WebRequestAuthOptions/signal}}"].
 
-          1. Otherwise:
+              1. Let |done| be `false`.
+
+              1. If |signal| is an {{AbortSignal}}, then [=AbortSignal/add=] the
+                  following abort algorithm to |signal|:
 
-              1. Set |result| to the result of calling |config|'s
-                  [=WebRequest handler config/handler=] given |details|.
+                  1. Set |result|'s [=WebRequestEvent result/cancel=] to `true`.
 
-                  Note: Any exceptions thrown by the handler are intentionally
-                  ignored and will not affect the request.
+                  1. Set |done| to `true`.
 
-          1. If all [=map/keys=] in |result| are [=list/contained=] in the set
-              « "cancel", "authCredentials" » and |result|["{{
-              BlockingResponse/authCredentials}}"] is either undefined or a
-              dictionary [=list/containing=] two [=map/keys=] equal to
-              "username" and "password", then:
+              1. [=Upon fulfillment=] of |credentials|, run the following steps
+                  that take a |resolvedValue|:
 
-              1. If |result|["{{BlockingResponse/cancel}}"] is `true`, then set
-                  |combinedResults|["{{BlockingResponse/cancel}}"] to `true`.
+                  1. Set |result|'s [=WebRequestEvent result/authCredentials=]
+                      to |resolvedValue|.
```

Comment:
_resolvedValue_ needs to be converted to a `WebRequestAuthCredentials` and this could fail.

Reply:
Done. I again don't cancel the request if this validation fails.